server/commands/write.go

```diff
@@ -104,3 +139,65 @@ func handleError(err error) error {
 
 	return serverErrors.HandleError("", err)
 }
+
+func isDirectIntersection(nodes *openfgapb.Userset_Intersection, tk *openfgapb.TupleKey) error {
+	isDirect := false
+contolLoop:
```

Comment:
Since Go [break statements](https://go.dev/ref/spec#Break_statements) act on `for`, `switch`, and `loop`, we need a named loop for the `for` loop here so that we can early exit if we find a direct relationship in any of the children

Reply:
Sorry, I just meant that I think it should be spelled `controlLoop`, not `contolLoop`. But actually, now that you are returning a bool here, I think you can just do `return true` on line 143, and remove the break entirely. WDYT?

Reply:
You are effectively using this to have a single point of exit in the function rather returning within the loop right?

Reply:
@craigpastro - 😂  I missed the typo completely... twice!

@craigpastro @elbuo8 - agree, now that the functions return `bool` not `err` these are unnecessary 🙌